CondCore/ESSources/interface/DataProxy.h

```diff
@@ -125,27 +128,24 @@ class DataProxyWrapper : public  cond::DataProxyWrapperBase {
     // when the plugin is dynamically loaded
     m_type = edm::eventsetup::DataKey::makeTypeTag<DataT>();
   }
-
-  // late initialize (to allow to load ALL library first)
-  void lateInit(cond::persistency::Session& session, const std::string & tag, const boost::posix_time::ptime& snapshotTime,
-			std::string const & il, std::string const & cs) override {
-    m_proxy.reset(new PayProxy(m_source.empty() ?  (const char *)nullptr : m_source.c_str() ) );
-    m_proxy->setUp( session );
-    m_proxy->loadTag( tag, snapshotTime );
-    m_edmProxy.reset(new DataProxy(m_proxy));
-    addInfo(il, cs, tag);
-  }
     
   edm::eventsetup::TypeTag type() const override { return m_type;}
-  ProxyP proxy() const override { return m_proxy;}
-  edmProxyP edmProxy() const override { return m_edmProxy;}
+  ProxyP proxy() const override { return m_proxy.get();}
+  edmProxyP makeEdmProxy() const override { 
+    assert(m_proxy.get() != nullptr);
+    return std::make_shared<DataProxy>(*m_proxy);}
  
 private:
+  // late initialize (to allow to load ALL library first)
+  void lateInitImpl(cond::persistency::Session& session, const std::string & tag, const boost::posix_time::ptime& snapshotTime) override {
+    m_proxy =std::make_unique<PayProxy>(m_source.empty() ?  (const char *)nullptr : m_source.c_str() );
+    m_proxy->setUp( session );
+    m_proxy->loadTag( tag, snapshotTime );
+  }
+
   std::string m_source;
   edm::eventsetup::TypeTag m_type;
-  std::shared_ptr<cond::persistency::PayloadProxy<DataT> >  m_proxy;
-  edmProxyP m_edmProxy;
-
+  std::unique_ptr<PayProxy>  m_proxy;
```

Comment:
Any real reason we need this indirection? PayloadProxy already contains a shared_ptr for the payload. Maybe adding full value semantics to PayloadProxy we could remove the unique_ptr?

Reply:
`lateInit` gets rid of the old object held by `m_proxy` and makes a new one. That would not be possible with value semantics.

Reply:
@ggovi how do you wish to proceed?

Reply:
Ok I see! Thanks.